tools/dumpobj.ml

```diff
@@ -89,9 +89,9 @@ let rec print_struct_const = function
   | Const_base(Const_string (s, _, _)) -> printf "%S" s
   | Const_immstring s -> printf "%S" s
   | Const_base(Const_char c) -> printf "%C" c
-  | Const_base(Const_int32 i) -> printf "%ldl" i
-  | Const_base(Const_nativeint i) -> printf "%ndn" i
-  | Const_base(Const_int64 i) -> printf "%LdL" i
+  | Const_base(Const_int32 (i, _)) -> printf "%ldl" i
+  | Const_base(Const_nativeint (i, _)) -> printf "%ndn" i
+  | Const_base(Const_int64 (i, _)) -> printf "%LdL" i
```

Comment:
In all other places where pretty-printing takes place, the original string formatting (the second tuple element) is used. I somewhat arbitrarily went for "normalisation to decimal" here, as this is "dumpobj", so the input is a binary, which means that we don't have the original formatting at hand anyway (I have not read the surrounding code in depth, so this is an assumption -- e.g. I'm not sure where the string representation actually comes from in this context).